plugins/org.eclipse.n4js.cli/src/org/eclipse/n4js/cli/compiler/N4jscCompiler.java

```diff
@@ -24,28 +27,39 @@
 import org.eclipse.n4js.cli.N4jscExitCode;
 import org.eclipse.n4js.cli.N4jscFactory;
 import org.eclipse.n4js.cli.N4jscOptions;
-import org.eclipse.n4js.ide.server.N4JSLanguageServerImpl;
-import org.eclipse.n4js.ide.server.N4JSWorkspaceManager;
+import org.eclipse.n4js.ide.xtext.server.DefaultBuildRequestFactory;
+import org.eclipse.n4js.ide.xtext.server.ProjectStatePersisterConfig;
+import org.eclipse.n4js.ide.xtext.server.XLanguageServerImpl;
+import org.eclipse.n4js.ide.xtext.server.XWorkspaceManager;
 import org.eclipse.n4js.smith.Measurement;
 import org.eclipse.n4js.smith.N4JSDataCollectors;
+import org.eclipse.n4js.tester.TestCatalogSupplier;
 import org.eclipse.xtext.workspace.IProjectConfig;
 
+import com.google.common.base.Stopwatch;
+import com.google.inject.Injector;
+
 /**
  * The entry point for all cli calls with the goal 'compile'
  */
 @SuppressWarnings("restriction")
 public class N4jscCompiler {
 	private final N4jscOptions options;
-	private final N4JSLanguageServerImpl languageServer;
+	private final XLanguageServerImpl languageServer;
 	private final N4jscLanguageClient callback;
-	private final N4JSWorkspaceManager workspaceManager;
+	private final XWorkspaceManager workspaceManager;
 
 	/** Starts the compiler in a blocking fashion */
 	static public void start(N4jscOptions options) throws Exception {
 		N4jscCompiler compiler = new N4jscCompiler(options);
 
 		try (Measurement m = N4JSDataCollectors.dcCliCompile.getMeasurement(options.toString())) {
+			SystemExitRedirecter.set();
+			System.out.println("before compiler.start()");
```

Comment:
I guess you are aware of all these debug outputs, but make sure to search the diff for "System.out" etc. and remove everything before merging to master.